datameta/resource.py

```diff
@@ -39,10 +39,11 @@ def resource_by_id(dbsession, model, idstring):
         entity = dbsession.query(model).filter(model.uuid==UUID(idstring)).one_or_none();
     except ValueError: # The specified string is not a valid UUID and the UUID constructor raises
         pass
-
-    try:
-        entity = entity if entity is not None else dbsession.query(model).filter(model.site_id==request_id).one_or_none();
-    except AttributeError: # This entity doesn't have a site_id
-        pass
+    
+    if hasattr(model, "site_id"):
```

Comment:
Whops, did I catch the wrong exception here?

Reply:
No, actually, when I tested it again, it worked fine, now.
Cannot remember why that solved a problem for me

Reply:
There is actually another bug there in main currently, maybe that was the problem...
https://github.com/ghga-de/datameta/blob/557b73b965ed4642bd2a9bd9485d54d6ddaad5e0/datameta/resource.py#L44
`request_id` is undefined here, should be `idstring` - maybe you ran into that?